src/posix/http_client_curl.cc

```diff
@@ -72,22 +84,32 @@ bool HTTPClientPlatformWrapper::RunHTTPRequest() {
   }
 
   cmd += url_requested_;
-  server_response_ = RunCurl(cmd);
-
-  // Clean up tmp file if any was created.
-  if (!post_body_.empty() && !post_file_.empty()) {
-    ::remove(post_file_.c_str());
+  try {
+    server_response_ = RunCurl(cmd);
+
+    // Clean up tmp file if any was created.
+    if (!post_body_.empty() && !post_file_.empty()) {
+      ::remove(post_file_.c_str());
+    }
+
+    error_code_ = -1;
+    std::string & s = server_response_;
+    if (s.size() < 3) {
```

Comment:
What is the hack? curl appends 3 chars of http code at the end, I want to avoid unnecessary allocations here so simply resize the string.
If pipe returned less than 3 chars it means that something went obviously wrong.

Anyway, it is not a full production-ready solution, of course. It should be improved in the future. But at least it works correctly now in simple cases.